algolia/search/index.go

```diff
@@ -96,3 +97,10 @@ func (i *Index) GetStatus(taskID int) (res TaskStatusRes, err error) {
 	err = i.transport.Request(&res, http.MethodGet, path, nil, call.Read)
 	return
 }
+
+// Exists returns whether an initialized index exists or not.
+func (i *Index) Exists() bool {
+	_, err := i.GetSettings()
+	_, ok := errs.IsAlgoliaErrWithCode(err, http.StatusNotFound)
+	return !ok
```

Comment:
Again, this implementation is just perfect if we want to consider network errors as `index does not exist` but according to what we discussed in person and what's already implemented in the PHP client: let's change a bit the signature here. We probably want to:

 1. have a return type of `(bool, error)` instead of just `(bool)`
 2. early return if the `GetSettings` returns a non-Algolia API error (you can check that thanks to the `errs.IsAlgoliaErr()` helper)

Reply:
Done 👍